eip-0008.md

```diff
@@ -0,0 +1,479 @@
+      validation itself is the fraction of this time (say 30 seconds), as the block must
+      also be distributed across the network and validated by all nodes. This is necessary
+      to avoid network splits.
+      - The maximum number of transactions in a block is limited by the `MaxBlockCost`
+      network parameter. This parameter can be increased by miners via voting (practically
+      indefinitely), as long as the resulting blocks don't exceed the validation time
+      limit.
+
+Under these limits, miners as a community, are interested to improve throughput of block
+assembly and validation software to process more transactions per second, with higher
+total cost within the given time limits. The proposed JIT costing protocol for extended
+UTXO model of Ergo enables a whole range of potential optimizations. This will be
+especially important when the emission is over and the transaction fees along with storage
+rent will be the only reward for miners.
+
+### JIT costing of ErgoTree
+
+Ethereum contracts are compiled to bytecode instructions, which are then executed by
+Ethereum VM as part of a transaction execution. Each instruction has an associated cost (aka
+'gas') and the cost of the whole transaction is accrued as instructions are executed one
+by one until either transaction finishes of gasLimit is reached, in which case the
+transaction is aborted and the `gasLimit * gasPrice` number of Wei is transferred to the
+miner.
+
+In contrast, Ergo doesn't use VM to execute contracts which protect boxes. The contract in
+Ergo is stored as serialized _typed abstract syntax tree_ (ErgoTree). The bytes of the
+contract are deserialized into ErgoTree data structure. 
+
+ErgoTree is designed for direct and efficient execution by the
+interpreter using a Context data structure to access the blockchain data. The interpreter
+doesn't have direct access to the blockchain, and rely only on the Context to execute ErgoTree.
+The original interpreter of the language is extended to
+accumulate the execution cost during evaluation of ErgoTree. Thus when the interpreter 
+finishes the execution of ErgoTree, the result is a pair `(R,C)`, where `R` is true/false
+boolean value and `C` is the cost accumulated during execution of `R`. If however during
+the execution of ErgoTree the accumulated cost exceeds at some point the `costLimit`
+value, then exception is thrown, which is interpreted as over-expensive script.
+
+An algorithm for execution of ErgoTree with JIT cost estimation is described in [ErgoTree
+Specification](https://ergoplatform.org/docs/ergotree.pdf).
+
+### Initial tests of transaction validity
+
+This conditions should be checked for every well-formed transaction.
+
+- All checks performed in Protocol v1
+- INPUTS contain at least one `p2pk(senderPk)` box
+- current height is within `feeOut.R6` interval
+- the `inputs` total amount and the `outputs` total amount are balanced
+- the `sum(collatInputs) + change == feeLimit` 
```

Comment:
why change is here?

Reply:
As describe in 3) of the protocol: "The
    `change` output box is balancing the total amount in `collatInputs` with `feeLimit` (see
    the figure)."